pkg/asset/cluster/cluster.go

```diff
@@ -79,7 +81,8 @@ func (c *Cluster) Generate(parents map[asset.Asset]*asset.State) (*asset.State,
 
 	stateFile, err := terraform.Apply(tmpDir, terraform.InfraStep, templateDir)
 	if err != nil {
-		return nil, err
+		// we should try to fetch the terraform state file.
```

Comment:
For cleanup?  For a later retry?  I'm fine just leaving the dead cluster and using the new tag/prefix cleanup you're adding to prune it.

Reply:
terrafrom always writes tfstate file, even if if was failed midway. It can be used for debugging to fix errors and use terraform to continue from where it left.

Reply:
> It can be used for debugging...

This sounds useful.

> ... use terraform to continue from where it left...

This sounds like something we don't want to support.  Hopefully install/destroy are both quick enough that the caller can just start clean.